### PR TITLE
OLS - drop missing

### PR DESCRIPTION
**Pull request by rjenc29**
New Feature
Basic mechanism to filter NaNs from features and targets, similar to statsmodels missing = 'drop'.  JIT compiles.

- [x] Code conforms to style guide
- [x] Doctests
- [x] Unit-tests
- [ ] Documentation
- [x] New line item in [releases.md](releases.md)
- [ ] Info added to `literature` directory

Bug Report

- [ ] Minimal repro in a test function

### CODE REVIEW

File: tests/linear_algebra/test_ols_sklearn_diabetes.py [hunk 1]
```diff
@@ -174,6 +179,109 @@ def test_f_statistic(self):
         assert np.allclose(output, expected)
```
[the remark this answers is not in the document]
Comment (dwillmer) [reply tests/linear_algebra/test_ols_sklearn_diabetes.py]: @codacy-bot ... you need some improvements. 

ignore this codacy issue report.

File: tests/linear_algebra/test_ols_sklearn_diabetes.py [hunk 2]
```diff
@@ -174,6 +179,109 @@ def test_f_statistic(self):
...
+        A, b, sm_model = self.statsmodels_test_fixtures()
+        output = fitted_values(*self.fn(A, b))
+        assert np.allclose(output, sm_model.fittedvalues)
+
+    def test_versus_statsmodels_residuals(self):
+        A, b, sm_model = self.statsmodels_test_fixtures()
+        output = residuals(*self.fn(A, b))
+        assert np.allclose(output, sm_model.resid)
+
+
```
[the remark this answers is not in the document]
Comment (dwillmer) [reply tests/linear_algebra/test_ols_sklearn_diabetes.py]: ignored

**Review by dwillmer — COMMENTED** [review]
Nice work - especially the xfail tests.

Seems to me like we should add these in with asserts and document the failure cases, with the intention of fixing later.

What's your opinion on the reasons for the failing tests?

**Comment by dwillmer** [comment]
This PR also looks good - just needs a line item and some discussion about why/how to use it in the docs (or docstrings) and we can merge for first release. Thanks!

File: tests/linear_algebra/test_ols_sklearn_diabetes.py [hunk 3]
```diff
@@ -174,6 +179,115 @@ def test_f_statistic(self):
...
+                           [1.2, 1.0, 1.3],
+                           [1.6, np.nan, 2.0],  # <- expect to be dropped
+                           [4.5, 4.2, 4.3],
+                           [4.4, 4.0, 4.2]])
+
+        self.b = np.array([1.0,
+                           6.0,
+                           2.0,
+                           3.0,
+                           np.nan])  # <- expect to be dropped
```
Comment (pawroman) [inline tests/linear_algebra/test_ols_sklearn_diabetes.py:195]: :+1:

**Review by pawroman — COMMENTED** [review]
Looks good, if anything - might be out of scope - I'd add a high-level usage guide for OLS.

File: fastats/linear_algebra/ols.py [hunk 4]
```diff
@@ -249,6 +249,42 @@ def f_statistic_no_intercept(A, b):
...
+
+    A_bar = np.empty_like(A)
+    b_bar = np.empty_like(b)
+
+    ctr = 0
+
+    for i in range(m):
+        feature_row = A[i, :]
+        target = b[i]
+        if np.all(~np.isnan(feature_row)) and ~np.isnan(target):
```
Comment (dwillmer) [inline fastats/linear_algebra/ols.py]: Any reason we can't use `np.all(np.isfinite(feature_row)) and not np.isnan(target)` ?

I think a finite check is more complete here, and i've never liked the `~` syntax for complement. :smile:

Comment (dwillmer) [reply fastats/linear_algebra/ols.py]: Perhaps ``np.isfinite(target)`` as well?

**Comment by dwillmer** [comment]
This is awesome, merged!

**Pull request merged**